test/service/test_manifest.py

```diff
@@ -437,14 +437,30 @@ def test_manifest_zarr(self):
                 self.assertEqual(200, response.status_code)
                 lines = response.content.decode().splitlines()
                 file_prefix = 'output="587d74b4-1075-4bbf-b96a-4d1ede0481b2/'
+                location_prefix = f'url="{config.service_endpoint()}/repository/files'
                 curl_files = []
+                urls = []
+                related_urls = []
                 for line in lines:
                     if line.startswith(file_prefix):
                         self.assertTrue(line.endswith('"'))
                         file_name = line[len(file_prefix):-1]
                         curl_files.append(file_name)
+                    elif line.startswith(location_prefix):
+                        self.assertTrue(line.endswith('"'))
+                        url = furl(line[len(location_prefix):-1])
+                        (related_urls if 'drsPath' in url.args else urls).append(url)
+                    else:
+                        # The manifest contains a combination of line formats,
+                        # we only validate `output` and `url` prefixed lines.
+                        pass
                 self.assertEqual(sorted([f['file_name'] for f in expected]),
                                  sorted(curl_files))
+                self.assertEqual(1, len(urls))
+                self.assertEqual(len(expected) - 1, len(related_urls))
+                expected_args = {'drsPath', 'fileName', 'requestIndex'}
+                for url in related_urls:
+                    self.assertSetEqual(expected_args - {*url.args.keys()}, set())
```

Comment:
I would have written 

```suggestion
                    self.assertSetEqual(expected_args - set(url.args.keys()), set())
```

because it's less cryptic.

Reply:
>… it's less cryptic.

Agreed, thanks for the suggestion.